joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/callgraph/MethodRefLinker.scala

```diff
@@ -78,24 +82,36 @@ object MethodRefLinker {
         srcNode
           .propertyOption(key)
           .filter { dstFullName =>
-            !srcNode.propertyDefaultValue(dstFullNameKey).equals(dstFullName)
+            !srcNode.propertyDefaultValue(dstFullNameKey).equals(dereferenceTypeFullName(dstFullName))
```

Comment:
```suggestion
            srcNode.propertyDefaultValue(dstFullNameKey) != dereferenceTypeFullName(dstFullName)
```

since we're already changing this line let's make it more scala-y

Reply:
Done.